### PR TITLE
WIP: Update to 2.17.0

### LINKED ISSUE #3: [RFE] Include Flash Player plugin in the Flatpak
Some meetings require Flash Player plugin to be installed and do not work if the plugin is not available.

Blue Jeans - Missing Flash Player error

### LINKED ISSUE #5: Crash closing "Report an issue" dialogue
I opened the "Report an issue" dialogue thinking I could use it to file an RFE but realised this was about reporting bugs. Closing the dialogue using the close button in the title bar crashed the application.

### LINKED ISSUE #14: Use BlueJeans v2
v1 (which we currently use) will stop working in a week (April 1st 2020).

Version 2.1.0 is available at: https://www.bluejeans.com/downloads

### PR DESCRIPTION
**Pull request by AsciiWolf**
Closes #14.
Closes #3.
Fixes #5.

Unfortunately, it does not work. BlueJeans starts with an empty window for some reason and `Unable to Create Log File` is printed on stdout, then I got message by abrt that bluejeans-v2 was killed by SIGSEGV. Strace found nothing suspicious. I will try further debugging it later.

Please see this comment: https://github.com/flathub/com.bluejeans.BlueJeans/pull/10#issuecomment-770840262

### CODE REVIEW

**Comment by ramcq** [comment]
I had a play with this, it's because different parts of the app are inconsistent as to whether they do or don't respect XDG_CONFIG_HOME. At least one place inside resources/app.asar and resources/start-detector.sh both hardcode ~/.config rather than checking the environment. strace shows:
```
95    access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs", F_OK) = 0
95    access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/", F_OK) = 0
104   openat(AT_FDCWD, "/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/bluejeans.log.03-02-2020", O_WRONLY|O_CREAT|O_APPEND|O_CLOEXEC, 0666) = 37
130   access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs", F_OK) = 0
130   access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/bluejeans-renderer.log.03-02-2020", F_OK) = 0
130   access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/bluejeans-renderer.log.03-02-2020.1", F_OK) = -1 ENOENT (No such file or directory)
130   stat("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/bluejeans-renderer.log.03-02-2020", {st_mode=S_IFREG|0644, st_size=0, ...}) = 0
130   access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/", F_OK) = 0
159   openat(AT_FDCWD, "/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/bluejeans-renderer.log.03-02-2020", O_WRONLY|O_CREAT|O_APPEND|O_CLOEXEC, 0666 <unfinished ...>
130   openat(AT_FDCWD, "/sysroot/home/ramcq/.config/bluejeans-v2/logs/BlueJeans-Linux-Media-2020-02-03T15-16-27[130].log", O_WRONLY|O_CREAT|O_APPEND, 0666) = 65
130   access("/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/", F_OK) = 0
130   openat(AT_FDCWD, "/sysroot/home/ramcq/.var/app/com.bluejeans.BlueJeans/config/bluejeans-v2/logs/.audit.json", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 53
```

Given we already have --persist=.config, running `mkdir -p ~/.config/bluejeans-v2/logs` does the trick, or I think you could possibly (even though it's redundant) force flatpak to do the same for you with `--persist=.config/bluejeans-v2` and `--persist=.config/bluejeans-v2/logs`.

**Comment by AsciiWolf** [comment]
@ramcq Thanks! Blue Jeans now launches without any issue, however when connected to a meeting, I cannot hear other people talking and I see a black screen instead of their video. Also, my camera does not seem to be detected and when I try to unmute my microphone, Blue Jeans crashes (shows empty blue screen).

By the way, there is a _BluejeansHelper_ binary that normally gets executed on login (Blue Jeans automatically creates a _~/.config/autostart/_ desktop file for it). However, even if I try running _BluejeansHelper_ manually before launching Blue Jeans, it seems to detect this helper process (according to the Blue Jeans stdout log), but nothing changes: Actual meeting in Blue Jeans is still not working.

**Comment by rothgar** [comment]
is the /build-repo/ path private? I was trying to test but can't load that uri to install.

I'd like to help to get this working if I can

**Comment by AsciiWolf** [comment]
@rothgar It isn't, it just needs a new build, the old one was already deleted.

bot, build

**Comment by rothgar** [comment]
Confirmed what you were seeing. It looks like my camera(s) aren't detected and xdg-open isn't working with this test build. (I'm on ubuntu 19.10)

I'm not familiar with building flatpaks but I'll see what I can try and get working.

**Comment by AsciiWolf** [comment]
There is a lot of EAGAIN in strace log:

```
poll([{fd=17, events=POLLIN}, {fd=18, events=POLLIN}, {fd=27, events=POLLIN}, {fd=40, events=POLLIN}], 4, 12) = 0 (Timeout)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
sendto(84, "\340\0\0\0\20\0\1\0\0\0\0\0\0\0\0\0\5\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 224, MSG_NOSIGNAL, NULL, 0) = 224
write(19, "!", 1)                       = 1
recvmsg(27, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(27, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=17, events=POLLIN}, {fd=18, events=POLLIN}, {fd=27, events=POLLIN}, {fd=40, events=POLLIN}], 4, 0) = 1 ([{fd=18, revents=POLLIN}])
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
read(18, "!!", 2)                       = 2
recvmsg(27, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(27, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=17, events=POLLIN}, {fd=18, events=POLLIN}, {fd=27, events=POLLIN}, {fd=40, events=POLLIN}], 4, 0) = 0 (Timeout)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(27, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(27, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(40, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
```

**Comment by AsciiWolf** [comment]
(The 4f035e5 and 6ef6f34 commits were not intentionally applied in this PR - full log output is useful when debugging the app. I will add them once the Blue Jeans app is working properly.)

File: com.bluejeans.BlueJeans.yaml [hunk 1]
```diff
@@ -53,28 +54,27 @@ modules:
...
-          - rpm2cpio bluejeans.rpm | cpio -idmv
-          - rm bluejeans.rpm
-          - mv opt/bluejeans/* .
-          - rm -r opt
-          - chmod 755 bluejeans
+          - rpm2cpio BlueJeans.rpm | cpio -idmv
+          - rm BlueJeans.rpm
+          - mv opt/BlueJeans/* .
+          - rm -r opt usr
+          - chmod 755 bluejeans-v2
```
Comment (hadess) [inline com.bluejeans.BlueJeans.yaml:68]: That's not needed, it's already executable

File: com.bluejeans.BlueJeans.yaml [hunk 2]
```diff
@@ -53,28 +54,27 @@ modules:
...
+          - "[ ! -d ~/.config/bluejeans-v2/logs ] && mkdir -p ~/.config/bluejeans-v2/logs"
+          - "exec /app/extra/bluejeans-v2 $@"
     build-commands:
       - install -D apply_extra /app/bin/apply_extra
       - install -D bluejeans /app/bin/bluejeans
-      - ARCH_TRIPLE="$(gcc --print-multiarch)" && ln -s "/usr/lib/${ARCH_TRIPLE}/libudev.so.1" /app/lib/libudev.so.0
       - install -Dm 644 com.bluejeans.BlueJeans.appdata.xml -t /app/share/appdata
       - install -Dm 644 com.bluejeans.BlueJeans.desktop -t /app/share/applications
-      - install -Dm 644 com.bluejeans.BlueJeans.png -t /app/share/icons/hicolor/128x128/apps
+      - install -Dm 644 com.bluejeans.BlueJeans.png -t /app/share/icons/hicolor/512x512/apps
```
Comment (hadess) [inline com.bluejeans.BlueJeans.yaml]: You need to update the png file from the RPM to be able to do that.

**Comment by hadess** [comment]
This is the backtrace of the crash, both with the 2.1.0 RPM and deb, and the 2.1.1 RPM, and also both RPM versions inside Flatpak:
```
#0  __strlen_avx2 () at ../sysdeps/x86_64/multiarch/strlen-avx2.S:65
#1  0x00007fffc68d5957 in std::char_traits<char>::length(char const*) (__s=0x7fffc6b21d96 "\003") at /usr/include/c++/5/bits/char_traits.h:267
#2  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&)
    (this=0x7fffc6cb17a0 <MediaLog::logFileName[abi:cxx11]>, __s=0x7fffc6b21d96 "\003", __a=...) at /usr/include/c++/5/bits/basic_string.h:456
#3  0x00007fffc68cff41 in __static_initialization_and_destruction_0(int, int) (__priority=65535, __initialize_p=1)
    at /home/denimuser/workspace/workspace/BFL_2.1.0-4ZBPICRPYAWVT7E3YEACQN3EGFJRLIREW544ENFOODYFNVNJ2ABQ/node_modules/fiber-wrapper-node/dependencies/checkouts/fiberclient-linux/source/simpleLogger.cpp:17
#4  0x00007ffff7fe1782 in call_init (l=<optimized out>, argc=argc@entry=1, argv=argv@entry=0x7fffffffd588, env=env@entry=0x3722afa29b80) at dl-init.c:72
#5  0x00007ffff7fe1891 in call_init (env=0x3722afa29b80, argv=0x7fffffffd588, argc=1, l=<optimized out>) at dl-init.c:30
#6  _dl_init (main_map=0x3722b0a32c00, argc=1, argv=0x7fffffffd588, env=0x3722afa29b80) at dl-init.c:119
#7  0x00007ffff6550ff5 in __GI__dl_catch_exception (exception=<optimized out>, operate=<optimized out>, args=<optimized out>) at dl-error-skeleton.c:182
#8  0x00007ffff7fe5928 in dl_open_worker (a=a@entry=0x7fffffffa470) at dl-open.c:758
#9  0x00007ffff6550f98 in __GI__dl_catch_exception (exception=<optimized out>, operate=<optimized out>, args=<optimized out>) at dl-error-skeleton.c:208
#10 0x00007ffff7fe514e in _dl_open
    (file=0x3722b06268f0 "/home/hadess/Projects/Flathub/com.bluejeans.BlueJeans/debian/opt/BlueJeans/resources/app.asar.unpacked/node_modules/fiber-wrapper-node/build/Release/fiberWrapper.node", mode=-2147483647, caller_dlopen=0x55555bc07333, nsid=-2, argc=1, argv=<optimized out>, env=0x3722afa29b80) at dl-open.c:837
#11 0x00007ffff7b1a39c in dlopen_doit (a=a@entry=0x7fffffffa690) at dlopen.c:66
#12 0x00007ffff6550f98 in __GI__dl_catch_exception (exception=exception@entry=0x7fffffffa630, operate=<optimized out>, args=<optimized out>) at dl-error-skeleton.c:208
#13 0x00007ffff6551063 in __GI__dl_catch_error (objname=0x3722afa09010, errstring=0x3722afa09018, mallocedp=0x3722afa09008, operate=<optimized out>, args=<optimized out>) at dl-error-skeleton.c:227
#14 0x00007ffff7b1abd9 in _dlerror_run (operate=operate@entry=0x7ffff7b1a340 <dlopen_doit>, args=args@entry=0x7fffffffa690) at dlerror.c:170
#15 0x00007ffff7b1a428 in __dlopen (file=<optimized out>, mode=<optimized out>) at dlopen.c:87
```

File: com.bluejeans.BlueJeans.yaml (hunk 1 above)
Comment (AsciiWolf) [reply com.bluejeans.BlueJeans.yaml:68]: It was not in older versions and the chmod command itself is harmless, so I think I will keep it there.

File: com.bluejeans.BlueJeans.yaml (hunk 2 above)
Comment (AsciiWolf) [reply com.bluejeans.BlueJeans.yaml]: Oops, I am dumb. Thanks, fixed.

**Comment by kenya888** [comment]
Hi. Thank you all for your effort of maintenance BJ package. Could you let me know how to access test build by bot? The instruction from flathubbot above doesnt't work for me (flatpakref file is not found)

**Comment by AsciiWolf** [comment]
@kenya888 The test build is automatically deleted after a few days. I will trigger a new build, however, as I said earlier, it doesn't work (seems to be a BlueJeans issue, not a Flatpak one).

bot, build

**Comment by kenya888** [comment]
@AsciiWolf Thank you. I got it!! Yes I understand it has some issues. I'd like to know what is happening on my desk. I'd like to find what i can do to fix issues:) Thank you.

**Comment by hadess** [comment]
It doesn't crash anymore! But my company's single-sign-on doesn't work, it does popup the "login" page in the browser, then shows that I can close the tab to carry on using the app, but the app hasn't reacted and the only option is cancelling the login process.

**Comment by ernestask** [comment]
> It doesn't crash anymore! But my company's single-sign-on doesn't work, it does popup the "login" page in the browser, then shows that I can close the tab to carry on using the app, but the app hasn't reacted and the only option is cancelling the login process.

I was being redirected to 127.0.0.1 instead for some reason. :confused:

**Comment by AsciiWolf** [comment]
> It doesn't crash anymore! But my company's single-sign-on doesn't work, it does popup the "login" page in the browser, then shows that I can close the tab to carry on using the app, but the app hasn't reacted and the only option is cancelling the login process.

Wait... The Flatpak version does not crash for you anymore? Hmm, weird, it still crashes for me when connecting to a meeting.

**Comment by hadess** [comment]
> Wait... The Flatpak version does not crash for you anymore? Hmm, weird, it still crashes for me when connecting to a meeting.

It used to crash on startup, now it doesn't. I never get to the point where it can connect to a meeting seeing as I can't login...

**Comment by AsciiWolf** [comment]
Ah, BlueJeans did not crash anymore since [this workaround](https://github.com/flathub/com.bluejeans.BlueJeans/pull/10/files#diff-3bc0a192b389abd89faa7bf7f8871241R73) was added. I did not test logging in using my company single-sign-on, I logged in as a guest, then connected to a meeting using its ID.

**Comment by tpopela** [comment]
@hadess @AsciiWolf I think that you will need something like https://github.com/flathub/com.getferdi.Ferdi/commit/b2f78604e98180b525184c50f75076726b6c4700 to get SSO working (the krb5 was dropped from the 19.08 runtime).

**Comment by hadess** [comment]
> @hadess @AsciiWolf I think that you will need something like [flathub/com.getferdi.Ferdi@b2f7860](https://github.com/flathub/com.getferdi.Ferdi/commit/b2f78604e98180b525184c50f75076726b6c4700) to get SSO working (the krb5 was dropped from the 19.08 runtime).

That wasn't enough I'm afraid. I have no idea how the SSO works, but I'm not actually logged in to the Red Hat Kerberos locally when using the 1.x version of the Bluejeans app, and I don't know what the web page is supposed to trigger in the app either.

Here's the patch I used:
[0001-Add-krb5-support.patch.txt](https://github.com/flathub/com.bluejeans.BlueJeans/files/4551967/0001-Add-krb5-support.patch.txt)

**Comment by AsciiWolf** [comment]
I have tried the latest official BlueJeans 2.1.2 rpm on Fedora 32 and it seems to work. So the crash when joining a meeting is a Flatpak version issue.

**Comment by AsciiWolf** [comment]
Still crashing when connecting to a meeting with microphone enabled. With microphone disabled, I can connect, but I can't hear or see anything.

**Comment by hadess** [comment]
> Still crashing when connecting to a meeting with microphone enabled. With microphone disabled, I can connect, but I can't hear or see anything.

With more recent versions of Flatpak, punching the `pulseaudio` hole in the sandbox [also opens up ALSA devices](https://github.com/flatpak/flatpak/blob/master/NEWS#L4). Could be a use of ALSA and a lack of error checking?

**Comment by hadess** [comment]
> With more recent versions of Flatpak

Version 1.7.3 in case this issue is visited later on:
```
Changes in 1.7.3
================

 * Allow direct ALSA device access if app has pulseaudio access.
[...]
```

**Comment by AsciiWolf** [comment]
I have Flatpak 1.6.3 on my system (Fedora 32).

**Comment by hadess** [comment]
> I have Flatpak 1.6.3 on my system (Fedora 32).

If you have a chance to test with this one:
https://koji.fedoraproject.org/koji/buildinfo?buildID=1522798

**Comment by AsciiWolf** [comment]
> If you have a chance to test with this one:
> https://koji.fedoraproject.org/koji/buildinfo?buildID=1522798

I unfortunately do not have any machine when I could install this. :-/

**Comment by hadess** [comment]
Didn't realise this couldn't be installed on F32. This should be though:
https://koji.fedoraproject.org/koji/taskinfo?taskID=45746286

**Comment by AsciiWolf** [comment]
Thanks, but I currently don't have any machine available that I would want to install a dev branch Flatpak package on. I want to keep my machines stable enough, especially the work ones. :-)

I will setup a VM when I have more time...

**Comment by hadess** [comment]
2.4.0 is out:
```
        url: https://swdl.bluejeans.com/desktop-app/linux/2.4.0/BlueJeans.rpm
        sha256: e0a14407e666e35f14cb503f6d30ec0c9d9dc9b12e6b6bbcb18e3f4b0a6210a0
        size: 72178164
```


But tested with the latest Flatpak, and still no go. The SSO/Kerberos redirection points to a port on 127.0.0.1 which isn't actually used (?) and the application's backend crashes when logging in to a meeting as a guest.

**Comment by hadess** [comment]
Still crashy as guest, still can't log in using SSO.

**Comment by hadess** [comment]
I also tested with version `2.17.0.11` which is available internally. It still crashes in `CallApiBase::startSIP()` when logging in as a guest, and the return host to the SSO is still a `127.0.0.1` URL that's not accessible outside the sandbox.

**Comment by kenya888** [comment]
> I also tested with version `2.17.0.11` which is available internally. It still crashes in `CallApiBase::startSIP()` when logging in as a guest, and the return host to the SSO is still a `127.0.0.1` URL that's not accessible outside the sandbox.

Thank you for sharing your test result. I'd like to share my investigation on F34(rawhide) silverblue with version `2.5.0.50` 

I found the `BlueJeansHelper` isn't running on flatpak environment which handle SSO with listening on 18171/tcp.  It should be copied to `~/.config/bluejeans-v2` and run by `start-detector.sh`, but it has copied to `$XDG_CONFIG_HOME(in my env /var/home/kenya888/.var/app/com.bluejeans.BlueJeans/config)/bluejeans-v2`. It would be better if it is fixed.

Here is just my idea for consideration. Can we discuss about the better way?
For example...

1. BJ app should copy `BlueJeansHelper` binary (and other resources) to `~/.config/bluejeans-v2`. The flatpak manifest is written enough to access host's $HOME/.config
https://github.com/flathub/com.bluejeans.BlueJeans/blob/master/com.bluejeans.BlueJeans.yaml#L18 Currently I don't find why it doesn't.

2. BJ app should check existence of `$XDG_CONFIG_HOME` environment variables, if it exists copy the `BlueJeansHelper` to `$XDG_CONFIG_HOME/bluejeans-v2`(set `$HOME/.config/bluejeans-v2` as fallback) and respect it as storing place of BJ App configs

By the way, after fixing this issue, the next one(callback URL and launching BJ App instance) is waiting for us... :cry:

**Comment by AsciiWolf** [comment]
bot, build

**Comment by AsciiWolf** [comment]
bot, build

**Comment by hadess** [comment]
I've passed the crash logs from the last Flatpak build to Bluejeans tech support today.

**Comment by hadess** [comment]
In addition to running the "detector" (`/app/extra/resources/BluejeansHelper`) using the script provided in the same path, you also need to add (replace?) the mimetypes handled in the .desktop file by `x-scheme-handler/bjnb` as done in the .desktop file in the RPM.

**Comment by hadess** [comment]
This makes SSO work (kind of), though it launches a second instance of the UI instead of reusing the already open one. Still asking for more info on how to fix it.
```diff
diff --git a/com.bluejeans.BlueJeans.desktop b/com.bluejeans.BlueJeans.desktop
index 63468b3..8f2cf2a 100644
--- a/com.bluejeans.BlueJeans.desktop
+++ b/com.bluejeans.BlueJeans.desktop
@@ -6,4 +6,4 @@ Terminal=false
 Type=Application
 Icon=com.bluejeans.BlueJeans
 Categories=Network;VideoConference;
-MimeType=x-scheme-handler/bjn;
+MimeType=x-scheme-handler/bjnb;
diff --git a/com.bluejeans.BlueJeans.yaml b/com.bluejeans.BlueJeans.yaml
index 5e3d404..540fb7d 100644
--- a/com.bluejeans.BlueJeans.yaml
+++ b/com.bluejeans.BlueJeans.yaml
@@ -66,12 +66,16 @@ modules:
           - mv opt/BlueJeans/* .
           - rm -r opt usr
           - chmod 755 bluejeans-v2
+          - chmod 755 /app/extra/resources/BluejeansHelper
       - type: script
         dest-filename: bluejeans
         commands:
           - "[ -d /var/home ] && unlink /home && mv /var/home /home"
           - "[ ! -d ~/.config/bluejeans-v2/logs ] && mkdir -p ~/.config/bluejeans-v2/logs"
-          - "exec /app/extra/bluejeans-v2 $@"
+          - "/app/extra/resources/BluejeansHelper &"
+          - "HELPER_PID=$!"
+          - "/app/extra/bluejeans-v2 $@"
+          - "kill -9 $HELPER_PID"
     build-commands:
       - install -D apply_extra /app/bin/apply_extra
       - install -D bluejeans /app/bin/bluejeans

```

**Comment by AsciiWolf** [comment]
@hadess Thanks, I have updated the PR.

**Comment by AsciiWolf** [comment]
It still crashes when connecting to a meeting, at least on my system. Also, it still does not seem to detect my webcam.

**Comment by A6GibKm** [comment]
By the way cpio 2.13 is part, of at least freedesktop runtime 20.08.

**Comment by AsciiWolf** [comment]
Unfortunately, the old BlueJeans app does not work anymore and the new one still does not work at all. If we won't get it working in next ~month, I will make the app EOL and ask Flathub admins to archive this repository.

**Comment by hadess** [comment]
It doesn't crash anymore! With version 2.19.0.61, it seems to actually start properly.

**Comment by hadess** [comment]
> It doesn't crash anymore! With version 2.19.0.61, it seems to actually start properly.

We just need to figure out a way to implement the single instance code, probably using a small D-Bus helper.

**Comment by hadess** [comment]
I've sent you my local changes including the aforementioned "small D-Bus helper" by mail. Let me know if there's anything else I can do to help.

**Comment by AsciiWolf** [comment]
Interesting, thanks! I will look at it in upcoming days.

**Comment by hadess** [comment]
> Also, it still does not seem to detect my webcam.

I've followed up internally here about what seems to be a lack of Pipewire support for webcam access, and screen sharing support.

**Comment by AsciiWolf** [comment]
To be honest, I am no longer interested in maintaining the BlueJeans Flatpak. There are too many issues with it and I have personally switched to using the browser version (that now also works on Fedora including Silverblue) instead. I am including the patch from @hadess as an attachment to this comment in case anyone wants to continue working on it (just ask someone from the [Flathub team](https://github.com/orgs/flathub/people) for push access to this repo). Thanks!

[0001-update.zip](https://github.com/flathub/com.bluejeans.BlueJeans/files/5904061/0001-update.zip)

**Comment by hadess** [comment]
Could you please give me access to the repo then?

**Comment by AsciiWolf** [comment]
@hadess I would like to, but I don't have rights to do it. You must ask @bilelmoussaoui @nedrichards or someone else from the Flathub team.

File: com.bluejeans.BlueJeans.yaml [hunk 3]
```diff
@@ -16,8 +16,9 @@ finish-args:
   - --socket=pulseaudio
   - --share=network
   - --persist=.config
-  - --persist=.macromedia
+  - --persist=.pki
   - --filesystem=xdg-videos
+  - --talk-name=org.freedesktop.Notifications
```
Comment (hadess) [inline com.bluejeans.BlueJeans.yaml:21]: Do you know what this was required by?

File: com.bluejeans.BlueJeans.yaml [hunk 4]
```diff
@@ -16,8 +16,9 @@ finish-args:
   - --socket=pulseaudio
   - --share=network
   - --persist=.config
-  - --persist=.macromedia
+  - --persist=.pki
```
Comment (hadess) [inline com.bluejeans.BlueJeans.yaml:19]: And do you know what this was required by?

Comment (AsciiWolf) [reply com.bluejeans.BlueJeans.yaml:19]: These directories are all created/used by the Blue Jeans client. The .macromedia one was used by Flash that is not present anymore in the BJ2 client. The .pki one is used by BJ2, probably to store some certificates/keys.

File: com.bluejeans.BlueJeans.yaml (hunk 3 above)
Comment (AsciiWolf) [reply com.bluejeans.BlueJeans.yaml:21]: Not sure, maybe desktop notifications from the BJ client?

**Comment by hadess** [comment]
I've redone most of the changes here into:
https://github.com/flathub/com.bluejeans.BlueJeans/pull/18

**Comment by AsciiWolf** [comment]
Thanks! Feel free to also delete the "update" branch when it is not needed anymore.

**Pull request closed without merging**